storybookjs/storybook
# Addon-docs: Add legacy transitional support for MDX1

Issue: #20145 



## What I did

See attached migration notes:

> If you get stuck with the [MDX2 upgrade](#mdx2-upgrade), we also provide opt-in legacy MDX1 support. This is intended as a temporary solution while you upgrade your Storybook; MDX1 will be discontinued in Storybook 8.0. The MDX1 library is no longer maintained and installing it results in `npm audit` security warnings.

Corresponding `mdx1-csf` PR here: https://github.com/storybookjs/mdx1-csf/pull/19/files

## How to test

1. Run a Webpack sandbox for template, e.g. `yarn task --task sandbox --start-from auto --template cra/default-ts`
2. Rename `Introduction.mdx` to `Introduction.stories.mdx`
3. Edit the file to contain an MDX1-only pattern, such as:

```
<style>{`
  .subheading {
    --mediumdark: '#999999';
  }
`}</style>
```

4. Observe Storybook fails to start
5. Follow the install instructions in the attached MIGRATION.md
6. Observe Storybook starts successfully and you can navigate to `Examples/Introduction` story

## Checklist



      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

#### Maintainers

      make sure to add the `ci:merged` or `ci:daily` GH label to it.

`["cleanup", "BREAKING CHANGE", "feature request", "bug", "documentation", "maintenance", "dependencies", "other"]`
